sw/airborne/modules/computer_vision/marker/detector.c

```diff
@@ -113,7 +113,7 @@ static void geo_locate_marker(struct Marker *marker, struct image_t *img) {
 
   // Scale the parameters based on distance to ground and focal point
   struct NedCoor_f *pos = stateGetPositionNed_f();
-  float agl = sonar_bebop.distance; // -pos->z
+  float agl = -pos->z; //sonar_bebop.distance;
```

Comment:
This is quite important distinction. I think the sonar distance is  more correct for what you want to do here. It is not pretty solution but it will probably do for now. Can you revert this change.